crypto/include/internal/evp_int.h

```diff
@@ -154,19 +154,27 @@ const EVP_MD *evp_keccak_kmac256(void);
  * object database.
  */
 int EVP_add_mac(const EVP_MAC *mac);
```

Comment:
Did I forget to remove that?

Reply:
Yup.

Reply:
There were cases where the digest/md/algorithm synonyms for the MD name param used by a MAC were not all supported.  I'm thinking that utility functions for this and the cipher/algorithm synonyms might be worthwhile.

Having this common processing in a single place will make changes easier and across the board.  The MACs and KDFs could both use the same utilities and some public key algorithms would also be able to.

Reply:
I assume that comment is misplaced.
For MACs, I simply transferred what was in the original EVP_PKEY implementations, they varied quite a bit. Personally, I would rather standardize on *one* name for each of those sets rather than globalizing all the varieties of each of those sets of names...

Reply:
How about we standardise on _algorithm_ for both ciphers and digests and have the PKEY wrappers do the necessary name substitution?  When the latter are eventually deprecated, we'll only have the one name.

The only problem would be if a future algorithm required both digests and ciphers simultaneously.

Reply:
> How about we standardise on _algorithm_ for both ciphers and digests and have the PKEY wrappers do the necessary name substitution?  When the latter are eventually deprecated, we'll only have the one name.

I *like* the idea

Reply:
... although, when we change applications like `opensssl pkeyutl`, we need to be careful what code we change.  If we have the `-kdf` option switch to use EVP_KDF rather than EVP_PKEY, it loses all support of the "legacy" names...  which might come as a surprise.  So let's just make a mental note that some of the stuff in such application should remain as they are.

Reply:
Good point.